consensus/miner_signalled_consensus.md

```diff
@@ -0,0 +1,200 @@
+# Miner Signalled Consensus Upgrade
+
+## Simple Summary
+
+This document describes the proposal of how nodes in the peer-to-peer network
+may activate a new hard-forking consensus protocol based on miner signalling.
+
+## Motivation
+
+Activating a hard-forking new consensus protocol based on miner signalling
+prevents diverging the block chain in case a large fraction of the miners have
+not upgraded their nodes to a version able to follow the new consensus protocol.
+
+## Specification
+
+### Overview
+
+On a node configured for a miner-signalled consensus upgrade,
+per chain fork,
+the new consensus protocol becomes active at a determined
+height if at least a determined majority of the key blocks in a determined
+"signalling" block interval in such chain fork contains a determined value in
+the header field `info`.
+
+The "signalling" block interval has a determined number of confirmations before
+the determined height at which the new consensus protocol may become
+active. This is in order to:
+- Prevent the node from having to aggregate the results of the
+  "signalling" block interval in a too short time interval.
+- Prevent a user from restarting the node configuring unconditionally
+  the new consensus protocol when the signalling outcome is not
+  confirmed with reasonably high probability.
+
+The value in the field `info` in the key block headers in the
+"signalling" block interval:
+- Uses the whole field `info` i.e. is not a bitmask.
+- Is meant to be a pseudorandom number, so to limit the risk of the same value
+  being used by simultaneous competing consensus protocol proposals.
+- Is not necessarily part of the validation rules of the new consensus
+  protocol.
+  For the sake of clarity: the miner signalling is orthogonal to the
+  validation rules of the new consensus protocol.
+  This means that, if the network resonably settles - via miner
+  signalling - on activating the new consensus protocol and a user
+  starts a node configuring the new consensus protocol as
+  unconditionally effective at the same height as the (past)
+  miner-signalled consensus upgrade height, the node may - depending
+  on the chain forks observed on the network - follow as best valid
+  chain a chain fork that upgraded to the new consensus protocol at
+  the configured height even if without a majority of "signals" in the
+  "signalling" block interval.
+
+### Parameters
+
+In order for the miner-signalled consensus protocol upgrade to succeed, each
+node must have the same set of parameters.
+
+Assumptions:
+- Let us call C1 the current consensus protocol version, and H1 the height at
+  which it was first effective.
+
+#### New consensus protocol version
+
+The proposed new consensus protocol version.
+
+It must be strictly greater than C1 (strictly monotonically increasing consensus
+protocol version).
+
+#### Signal
+
+The "signal" value in the block header field `info`.
+
+It should be pseudorandom.
```

Comment:
what does that mean?

Reply:
It's to make sure that in case of concurrent forks - meaning that aeternity will release a node which will try to upgrade to a new protocol, but also another party will do the same so the two signals will not be the same.